### ISSUE TITLE
[Rule change] avoid-global-state to support static fields

### ISSUE THREAD

**Issue by ookami-kb** [dart-code-checker/dart-code-metrics#643]
**What rule do you want to change?**

`avoid-global-state`

**How will the change be implemented? (New option, new default behavior, etc.)?**

According to the documentation example, this code should trigger the rule:

```dart
class Foo {
  static int? bar; // LINT
}
```

As discussed in tg group, this is currently not implemented but it's on the roadmap. Creating the issue to keep track of it.

**Please provide some example code that this change will affect:**

```dart
class Foo {
  static int? bar; // LINT
}
```

**What does the rule currently do for this code?**

Nothing.

**What will the rule do after it's changed?**

Trigger rule.

**Are you willing to submit a pull request to implement this change?**

Probably.

**Comment by dkrutskikh** [comment]
Available in 4.9.1 release 🚀
